client.go

```diff
@@ -467,6 +388,24 @@ func (co *Conn) Write(p []byte) (n int, err error) {
 	return n, err
 }
 
+// Return the appropriate timeout for a specific request
+func (c *Client) getTimeoutForRequest(timeout time.Duration) time.Duration {
+	var requestTimeout time.Duration
+	if c.Timeout != 0 {
+		requestTimeout = c.Timeout
+	} else {
+		requestTimeout = timeout
+	}
+	// net.Dialer.Timeout has priority if smaller than the timeouts computed so
```

Comment:
Is this documented anywhere else?

Reply:
Yes, at line 35 (the one after the long one that I will break in the next commit):

```
DialTimeout    time.Duration     // net.DialTimeout, defaults to 2 seconds, or net.Dialer.Timeout if
expiring earlier - overridden by Timeout when that value is non-zero
```

While I also updated `docs.go`, I didn't call this out explicitly to stay true to the style of the rest of the document (i.e. no information overload). The current text there is:

```
More advanced options are availabe using a net.Dialer and the corresponding API.
For example it is possible to set a timeout, or to specify a source IP address
and port to use for the connection:

        c := new(dns.Client)
        laddr := net.UDPAddr{
                IP: net.ParseIP("[::1]"),
                Port: 12345,
                Zone: "",
        }
        d := net.Dialer{
                Timeout: 200 * time.Millisecond,
                LocalAddr: &laddr,
        }
        in, rtt, err := c.ExchangeWithDialer(&d, m1, "8.8.8.8:53")
```